src/actions/index.js

```diff
@@ -68,7 +67,8 @@ function fetchDirections() {
     if (language) options.push('language='+language);
     if (exclude) options.push('exclude=' + exclude);
     if (accessToken) options.push('access_token=' + accessToken);
-    request.abort();
+
+    const request = new XMLHttpRequest();
```

Comment:
Probably after that will need to follow order of requests in callbacks to have ability anyway cancel requests and don't trigger long response updates

Reply:
[6b80678](https://github.com/mapbox/mapbox-gl-directions/pull/310/commits/6b80678fdf529ff43e7a686a3fcd1868224aaa1e)